cpp/llm_chat.cc

```diff
@@ -449,7 +449,7 @@ class LLMChat {
       this->sliding_window_size_ = config["sliding_window_size"].get<int64_t>();
       CHECK(this->sliding_window_size_ > 0 || this->sliding_window_size_ == -1)
           << "Sliding window size needs to be -1 or positive";
-      CHECK(config.count("prefill_chunk_size"))
+      CHECK(config.count("prefill_chunk_size") || this->sliding_window_size_ == -1)
```

Comment:
We should always require `prefill_chunk_size` to exist as a positive integer IIUC

Reply:
My `gen_config` results in -1 for both `prefill_chunk_size` and `sliding_window_size`:

```
{
  "model_type": "llama",
  "quantization": "q0f16",
  "model_config": {
    ...
  },
  "vocab_size": 32000,
  "context_window_size": 4096,
  "sliding_window_size": -1,
  "prefill_chunk_size": 4096,
  "attention_sink_size": -1,
  ...,
  "version": "0.1.0"
}
```

Reply:
We assume `prefill_chunk_size` should be a meaningful positive integer that's bounded by `[1, conetxt_window_size/sliding_window_size]`, and if not, we would consider it a bug. Another assumption is that one of `context_window_size` and `sliding_window_size` should be -1, and the other be a meaningful positive integer.

Reply:
Sorry for the confusion. I must have made some mistake in understanding before. Just reverted this change :-)